### PR TITLE
Add static pose publisher and support pose_v msg type in pose_publisher system

### PR DESCRIPTION
**Pull request by iche033**
Added two features to help reduce bandwidth of pose-publisher's pose topic

* Static pose publisher:
    * When the `<static_publisher>` sdf element is set to true, a static pose publisher is created that publishes static pose transform for entities in the model that do not change pose. Links connected by joints are considered as dynamic entities (except for canonical link) and their pose is published to the original `/model/[model_name]/pose` topic, while the rest are published to the new `/model/[model_name]/pose_static` topic.  
    * Ideally the static pose publisher should be latched like the `/tf_static` topic so we don't have to keep publishing the data but I don't think ign transport has latched support yet. So I added  `<static_update_frequency>` sdf element to let users configure / reduce the publisher's update rate 

* Pose_V message:
    * Added the `<use_pose_vector_msg>` sdf element to support publishing `ignition::msgs::Pose_V` msgs (vector of poses at one sim iteration) as opposed to individual `ignition::msgs::Pose msgs`. This helps to reduce the number of published msgs with the trade-off of larger msg  size. The default pose publisher's publish rate should now be equal to the simulation's update rate, whereas before the update rate was equal to num_entities_in_model * sim_update_rate

related [ros_ign pull request](https://github.com/osrf/ros_ign/pull/67)

### CODE REVIEW

**Comment by chapulina** [comment]
@osrf-jenkins run tests

File: src/systems/pose_publisher/PosePublisher.cc [hunk 1]
```diff
@@ -91,27 +114,53 @@ class ignition::gazebo::systems::PosePublisherPrivate
...
   /// member variable to avoid repeated memory allocations and improve
   /// performance.
   public: ignition::msgs::Pose poseMsg;
 
+  /// \brief A variable that gets populated with poses. This also here as a
+  /// member variable to avoid repeated memory allocations and improve
+  /// performance.
+  public: ignition::msgs::Pose_V poseVMsg;
+
+
```
Comment (mjcarroll) [inline src/systems/pose_publisher/PosePublisher.cc:159]: Nit: extra vertical whitespace

File: src/systems/pose_publisher/PosePublisher.cc [hunk 2]
```diff
@@ -277,38 +443,79 @@ void PosePublisherPrivate::InitializeEntitiesToPublish(
...
+  if (this->usePoseV)
+    this->poseVMsg.Clear();
+  for (const auto &[entity, pose] : _poses)
   {
     auto entityIt = this->entitiesToPublish.find(entity);
     if (entityIt == this->entitiesToPublish.end())
       continue;
+
+    if (this->usePoseV)
+      msg = this->poseVMsg.add_pose();
```
Comment (mjcarroll) [inline src/systems/pose_publisher/PosePublisher.cc:507]: Maybe curly braces around one line of logic.

File: src/systems/pose_publisher/PosePublisher.cc [hunk 3]
```diff
@@ -277,38 +443,79 @@ void PosePublisherPrivate::InitializeEntitiesToPublish(
...
+      msg = &this->poseMsg;
+    }
+
     // fill pose msg
-    this->poseMsg.Clear();
     // frame_id: parent entity name
     // child_frame_id = entity name
     // pose is the transform from frame_id to child_frame_id
-    auto header = this->poseMsg.mutable_header();
+    auto header = msg->mutable_header();
```
Comment (mjcarroll) [inline src/systems/pose_publisher/PosePublisher.cc:518]: Is there any chance of `msg` being null here?

File: test/integration/pose_publisher_system.cc [hunk 4]
```diff
@@ -390,3 +417,323 @@ TEST_F(PosePublisherTest, UpdateFrequency)
...
+      {
+        FAIL() << "Unknown link found: " << msg.name();
+      }
+      EXPECT_EQ(expectedPose, p);
+    }
+    timestamps.pop_front();
+  }
+  mutex.unlock();
+
+
```
Comment (mjcarroll) [inline test/integration/pose_publisher_system.cc:618]: Nit: extra vertical whitespace

File: test/worlds/pose_publisher.sdf [hunk 5]
```diff
@@ -268,20 +268,253 @@
...
         name="ignition::gazebo::systems::PosePublisher">
         <publish_link_pose>true</publish_link_pose>
+        <publish_sensor_pose>true</publish_sensor_pose>
+        <publish_collision_pose>false</publish_collision_pose>
+        <publish_visual_pose>false</publish_visual_pose>
+        <publish_nested_model_pose>false</publish_nested_model_pose>
+        <static_publisher>true</static_publisher>
+        <use_pose_vector_msg>true</use_pose_vector_msg>
+      </plugin>
+
```
Comment (mjcarroll) [inline test/worlds/pose_publisher.sdf:485]: Extra whitespace

File: src/systems/pose_publisher/PosePublisher.cc (hunk 1 above)
Comment (iche033) [reply src/systems/pose_publisher/PosePublisher.cc]: removed. 4255f22

File: src/systems/pose_publisher/PosePublisher.cc (hunk 2 above)
Comment (iche033) [reply src/systems/pose_publisher/PosePublisher.cc:508]: added. 4255f22

File: src/systems/pose_publisher/PosePublisher.cc (hunk 3 above)
Comment (iche033) [reply src/systems/pose_publisher/PosePublisher.cc:521]: it should never be null. Added `IGN_ASSERT` to be sure. 4255f22

File: test/integration/pose_publisher_system.cc (hunk 4 above)
Comment (iche033) [reply test/integration/pose_publisher_system.cc]: removed. 4255f22

File: test/worlds/pose_publisher.sdf (hunk 5 above)
Comment (iche033) [reply test/worlds/pose_publisher.sdf]: removed. 4255f22

**Pull request merged**